### PR TITLE
feat: display maintenance page in case of a 503 http status error 

### PR DESCRIPTION
**Pull request by SGrueber**
## PR Type

[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no API changes)
[ ] Build-related changes
[ ] CI-related changes
[ ] Documentation content changes
[ ] Application / infrastructure changes
[ ] Other: 

## What Is the Current Behavior?
If the ICM server is in maintenance mode REST requests respond with an 503 http status code, but the client doesn`t handle these errors properly.

Issue Number: Closes #

## What Is the New Behavior?
The pwa shows a maintenance page if a REST request respond with a 503 error.

## Does this PR Introduce a Breaking Change?

[ ] Yes
[x] No

### ICM 7.10 
Webadapter (WA) 1.1.4 or higher
### ICM 11 
Webadapter (WA) 2.4.3 or higher

[AB#75906](https://dev.azure.com/intershop-com/cefd1005-00a7-4c79-927f-a16947d1b2e6/_workitems/edit/75906)

### CODE REVIEW

**Review by shauke — CHANGES_REQUESTED** [review]

**Comment by jometzner** [comment]
@shauke @SGrueber We can now retest the integration with ICM again, as the WA is now capable of handling CORS preflight requests in the maintenance case.

**Review by shauke — CHANGES_REQUESTED** [review]

File: src/assets/i18n/fr_FR.json [hunk 1]
```diff
@@ -1043,7 +1043,13 @@
   "seo.title.checkout": "Passer la commande",
   "seo.title.error": "Erreur",
   "seo.title.home": "Page d’accueil InTRONICS",
+  "seo.title.maintenance": "Maintenance",
   "seo.title.search": "Résultat de recherche pour '{{0}}'.",
+  "server.maintenance.page.contact.email": "info@test.intershop.de",
+  "server.maintenance.page.contact.label": "You can contact us:",
+  "server.maintenance.page.contact.phone": "1-800-xxx-xxxx",
+  "server.maintenance.page.text": "Nous sommes désolés ...</h3><h4></h4>",
```
[the remark this answers is not in the document]
Comment (shauke) [reply src/assets/i18n/fr_FR.json:1051]: @mglatter & @NicolasGrizard The other localizations should be reviewed as well. Thank you.

File: src/app/shell/header/header-error/header-error.component.html [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<div class="header header-error container">
+  <div class="mid-header">
+    <div class="logo-wrapper d-none d-md-block">
+      <a
+        rel="home"
+        href="/home"
```
Comment (jometzner) [inline src/app/shell/header/header-error/header-error.component.html:6]: This is not working in a prod like deployment with multi-channel basehref configuration

**Review by jometzner — APPROVED** [review]

**Review by shauke — DISMISSED** [review]

File: src/assets/i18n/fr_FR.json [hunk 3]
```diff
@@ -1043,7 +1043,13 @@
   "seo.title.checkout": "Passer la commande",
   "seo.title.error": "Erreur",
   "seo.title.home": "Page d’accueil InTRONICS",
+  "seo.title.maintenance": "Maintenance",
   "seo.title.search": "Résultat de recherche pour '{{0}}'.",
+  "server.maintenance.page.contact.email": "info@test.intershop.de",
+  "server.maintenance.page.contact.label": "You can contact us:",
```
Comment (NicolasGrizard) [inline src/assets/i18n/fr_FR.json:1049]: should maybe be in french: "Vous pouvez nou contacter:",

File: src/assets/i18n/fr_FR.json [hunk 4]
```diff
@@ -1043,7 +1043,13 @@
...
+  "server.maintenance.page.contact.phone": "1-800-xxx-xxxx",
+  "server.maintenance.page.text": "Nous sommes désolés ...</h3><h4></h4>",
+  "server.maintenance.page.title": "Arrêts de maintenance",
```
Comment (NicolasGrizard) [inline src/assets/i18n/fr_FR.json:1052]: it should be "arrêt", no plural s

**Review by NicolasGrizard — CHANGES_REQUESTED** [review]
see both comments

**Review by shauke — APPROVED** [review]

**Pull request merged**